### PR TITLE
Simplifying and fixing fix_batch_norm_names

### PR DESCRIPTION
**Pull request by corey-nm**
In torch 1.12, some of the initializers are used by multiple nodes. For this fix_batch_norm_names function, this was causing a bug where:
1. A node's input & initializer were updated by removing ".module"
2. Another node that also used that initializer was not updated because the check for `init_name not in name_to_inits` failed, so the node was skipped

This PR simplifies the logic of the above by changing initializer names and node input/output names separately.

### CODE REVIEW

**Review by KSGulin — APPROVED** [review]

File: src/sparseml/pytorch/utils/exporter.py [hunk 1]
```diff
@@ -745,25 +745,15 @@ def _delete_trivial_onnx_adds(model: onnx.ModelProto):
             continue
 
 
-def _fix_batch_norm_names(model: onnx.ModelProto):
-    name_to_inits = {init.name: init for init in model.graph.initializer}
+def _remove_module_from_names(model: onnx.ModelProto):
+    onnx.checker.check_model(model)
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:749]: why do we need to check before?

File: src/sparseml/pytorch/utils/exporter.py [hunk 2]
```diff
@@ -745,25 +745,15 @@ def _delete_trivial_onnx_adds(model: onnx.ModelProto):
...
+
+    for init in model.graph.initializer:
+        init.name = init.name.replace(".module", "")
     for node in model.graph.node:
-        if node.op_type != "BatchNormalization":
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:751]: would prefer to keep scope still just to BN ops, `module` is still a valid name

File: src/sparseml/pytorch/utils/exporter.py (hunk 1 above)
Comment (corey-nm) [reply src/sparseml/pytorch/utils/exporter.py:749]: for more checks 😁 i can remove it

File: src/sparseml/pytorch/utils/exporter.py (hunk 2 above)
Comment (corey-nm) [reply src/sparseml/pytorch/utils/exporter.py:751]: and i guess only initializers that go into batch norms?

Comment (corey-nm) [reply src/sparseml/pytorch/utils/exporter.py:751]: Okay I changed the name `module` to something else that should be way more specific to this wrapper. Prefer to keep the simpler logic of this function because the previous version was harder to understand than it needs to be

**Review by bfineran — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]
